learning_resources/etl/canvas.py

```diff
@@ -58,11 +59,14 @@ def sync_canvas_archive(bucket, key: str, overwrite):
         )
         checksum = calc_checksum(course_archive_path)
         if run:
-            load_content_files(
-                run,
+            canvas_content_files = list(
```

Comment:
This seems to defeat the purpose of `transform_canvas_content_files` yielding results one at a time instead of returning everything at once, and could potentially cause memory issues for archives containing lots of contentfiles.  My first reaction was to revert it, but I get different counts with this code (139) vs as it was (82).  Not sure why, any ideas?

Reply:
i stumbled on the same quirk. I initially thought it had to do with the file handlers closing prematurely but that doesnt seem to be the case. Ill see if i can suss it out.

Reply:
fixed